File: summer/WebContent/js/system/windows/input/FocusNavigationDirection.js
/**
 * FocusNavigationDirection
 */
define(["dojo/_base/declare", "system/Type"], function(declare, Type){
	
    var FocusNavigationDirection = declare("FocusNavigationDirection", Object, { 
    });	
    
    FocusNavigationDirection.Next = 0;

    FocusNavigationDirection.Previous=1; 

    FocusNavigationDirection.First=2;
    
    FocusNavigationDirection.Last = 3;

    FocusNavigationDirection.Left=4; 

    FocusNavigationDirection.Right=5;
    
    FocusNavigationDirection.Up=6;
    
    FocusNavigationDirection.Down=7;

//    FocusNavigationDirection.Type = new Type("FocusNavigationDirection", FocusNavigationDirection, [Object.Type]);
    return FocusNavigationDirection;
});	

File: org.summer.view.widget/src/org/summer/view/widget/input/FocusNavigationDirection.java
package org.summer.view.widget.input;

public enum FocusNavigationDirection {
	Down,//	Move focus to another focusable element downwards from the currently focused element. 
	First,//	Move focus to the first focusable element in tab order. Not supported for PredictFocus. 
	Last,//	Move focus to the last focusable element in tab order. Not supported for PredictFocus. 
	Left,//	Move focus to another focusable element to the left of the currently focused element. 
	Next,//	Move focus to the next focusable element in tab order. Not supported for PredictFocus. 
	Previous,//	Move focus to the previous focusable element in tab order. Not supported for PredictFocus. 
	Right,//	Move focus to another focusable element to the right of the currently focused element. 
	Up,//	Move focus to another focusable element upwards from the currently focused element. 
}
